scripts/bootstrap.sh.in

```diff
@@ -161,27 +164,69 @@ die() {
     return 1
 }
 
-check_package_managers() {
-    case $OS in
-        centos)
-            test -n "$RPM" || die "rpm not found"
-            test -x "$RPM" || die "rpm at '$RPM' is not executable"
-            test -n "$YUM" || die "yum not found"
-            test -x "$YUM" || die "yum at '$YUM' is not executable"
-            ;;
-        ubuntu)
-            test -n "$DPKG" || die "dpkg not found"
-            test -x "$DPKG" || die "dpkg at '$DPKG' is not executable"
-            test -n "$APT" || die "apt not found"
-            test -x "$APT" || die "apt '$APT' is not executable"
-            ;;
-        *)
-            die "OS $OS not supported";;
-    esac
+check_package_manager_yum() {
+    test -n "$RPM" || die "rpm not found"
+    test -x "$RPM" || die "rpm at '$RPM' is not executable"
+    test -n "$YUM" || die "yum not found"
+    test -x "$YUM" || die "yum at '$YUM' is not executable"
+}
+
+check_package_manager_apt() {
+    test -n "$DPKG" || die "dpkg not found"
+    test -x "$DPKG" || die "dpkg at '$DPKG' is not executable"
+    test -n "$APT" || die "apt not found"
+    test -x "$APT" || die "apt '$APT' is not executable"
+}
+
+get_packages_list() {
+    python -c "
+import json
+
+with open('$BASE_DIR/salt/metalk8s/versions.json', 'r') as fd:
+    versions = json.load(fd)
+
+packages = versions.get('packages', {}).get('$OS', {})
+for pkg, pkg_info in packages.items():
+    print('{0}{1}'.format(
+        pkg, '-' + pkg_info['version'] if pkg_info.get('version') else '')
+    )
+"
+}
+
+check_rpm_packages() {
+    local -a packages
+
+    if read -ra packages < <(get_packages_list); then
+        yum install --assumeyes --setopt tsflags=test "${packages[@]}" || \
+            die "some packages or dependencies are missing in the configured" \
+                "repositories"
+    fi
+}
+
+check_dist_centos() {
+    check_package_manager_yum
+    check_rpm_packages
+}
+
+check_dist_redhat() {
+    check_package_manager_yum
+    command -v subscription-manager || die "subscription-manager not found"
+    subscription-manager status || \
+        die "system is not registered to any subscription management service"
```

Comment:
Do we want to fail if we are not registered ? Because we may have all repo locally (even if it's not a good way of support redhat it's doable)

Reply:
Hum, I don't know... maybe we could raise a warning message instead.
But to me, it's weird to use RHEL and not RHSM.

Reply:
Just checked for offline machine this command return 1 even if the machine is well registered (because not able to check the status)
https://access.redhat.com/solutions/3121571
```
[root@test-rhel ~]# subscription-manager status
+-------------------------------------------+
   System Status Details
+-------------------------------------------+
Overall Status: Unknown

[root@test-rhel ~]# echo $?
1
```

Reply:
Well, since we want to install packages under the realm of `rhsm`, shouldn't we explicitly request to have a (optionally proxied) connection to RedHat services? Or is it achievable/cautionable to install supported packages without validating subscriptions?